### PR TITLE
SNOW-824308 support bdate_range

### PR DESCRIPTION
**Pull request by sfc-gh-azhan**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-824308 support bdate_range

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/snowpark/modin/pandas/general.py [hunk 1]
```diff
@@ -2714,6 +2716,104 @@ def date_range(
...
+    end=None,
+    periods: int | None = None,
+    freq: Frequency | dt.timedelta = "B",
+    tz=None,
+    normalize: bool = True,
+    name: Hashable | None = None,
+    weekmask=None,
+    holidays=None,
+    inclusive: IntervalClosedType = "both",
+    **kwargs,
```
Comment (sfc-gh-nkrishna) [inline src/snowflake/snowpark/modin/pandas/general.py:2730]: Can we add type annotations here for all of the params? e.g. `tz: str | None = None`

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py [hunk 2]
```diff
@@ -726,6 +733,8 @@ def from_date_range(
             dt_values = ns_values.series_to_datetime()
 
         dt_series = pd.Series(query_compiler=dt_values)
+        if remove_non_business_days:
+            dt_series = dt_series[dt_series.dt.dayofweek < 5]
```
Comment (sfc-gh-nkrishna) [inline src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:737]: This seems to be removing just weekends, perhaps change the flag to `remove_non_business_days_excluding_holidays`? I'm not sure if there is a pandas function (outside of using the `CustomBusinessDay` param) that excludes holidays as well though so maybe this isn't necessary?

File: src/snowflake/snowpark/modin/plugin/_internal/generator_utils.py [hunk 3]
```diff
@@ -212,7 +223,12 @@ def generate_irregular_range(
         while start + periods * offset <= end:
             periods += 1
 
-    num_offsets = get_active_session().range(start=0, end=periods, step=1)
+    # TODO: SNOW-1646883 fix invalid identifier error when sql_simplifier_enabled is True
+    session = get_active_session()
+    sql_simplifier_enabled = session.sql_simplifier_enabled
+    session.sql_simplifier_enabled = False
+    num_offsets = session.range(start=0, end=periods, step=1)
+    session.sql_simplifier_enabled = sql_simplifier_enabled
```
Comment (sfc-gh-nkrishna) [inline src/snowflake/snowpark/modin/plugin/_internal/generator_utils.py:231]: Can we add a comment for why we disable the sql_simplifier for using `session.range`?

**Review by sfc-gh-nkrishna — COMMENTED** [review]
Some small comments

File: src/snowflake/snowpark/modin/plugin/_internal/generator_utils.py (hunk 3 above)
Comment (sfc-gh-azhan) [reply src/snowflake/snowpark/modin/plugin/_internal/generator_utils.py:231]: see line 226

Comment (sfc-gh-nkrishna) [reply src/snowflake/snowpark/modin/plugin/_internal/generator_utils.py:231]: Oh missed that, resolving this comment

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py (hunk 2 above)
Comment (sfc-gh-azhan) [reply src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:737]: pandas business day by default only means weekends but not holidays. We could revisit this if we decide to support CustomBusinessDay in the future IMO.

**Review by sfc-gh-nkrishna — APPROVED** [review]

**Review by sfc-gh-nkumar — APPROVED** [review]

**Pull request merged**